pkg/controllers/provisioning/provisioner.go

```diff
@@ -116,37 +115,56 @@ func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (resul
 	if len(machines) == 0 {
 		return reconcile.Result{}, nil
 	}
-	machineNames, err := p.LaunchMachines(ctx, machines, RecordPodNomination)
-
-	// Any successfully created node is going to have the nodeName value filled in the slice
-	successfullyCreatedNodeCount := lo.CountBy(machineNames, func(name string) bool { return name != "" })
-	metrics.NodesCreatedCounter.WithLabelValues(metrics.ProvisioningReason).Add(float64(successfullyCreatedNodeCount))
+	nodes, err := p.LaunchMachines(ctx, machines, RecordPodNomination)
+	p.UpdateLaunchMetrics(nodes, metrics.ProvisioningReason)
 
 	return reconcile.Result{}, err
 }
 
+func (p *Provisioner) UpdateLaunchMetrics(nodes []*v1.Node, reason string) {
```

Comment:
I'm not convinced that we need a separate function to do this. What if we just updated the metric in the `Launch` call and used the node label to update the metric?

i.e.
```go
metrics.NodesCreatedCounter.With(prometheus.Labels{
	metrics.ReasonLabel:      metrics.ProvisioningReason,
	metrics.ProvisionerLabel: k8sNode.Labels[v1alpha5.ProvisionerNameLabelKey],
}).Add(float64(nodeCount))
```

Reply:
I don't see a problem with moving that.  Just to make sure - are we ok with passing the provisioning reason to `Launch` ? (depending if it is a provisioning or replacement (deprovisioning) launch)

Reply:
I think the problem is that when you add functions like this that can be done inline it creates additional indirection that someone has to follow.